File: 백준/Silver/10828. 스택/스택.java
import java.util.*;
import java.io.*;

public class Main {
    static ArrayList<Integer> stack = new ArrayList<Integer>();
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());

        for (int i=0; i<T; i++) {
            String input = br.readLine();
            int lastIndex = stack.size()-1;

            if ( input.startsWith("push") ) {
                String[] inputArr = input.split(" ");
                int inputNum = Integer.parseInt(inputArr[1]);

                stack.add(inputNum);
            }

            else if ( input.equals("pop") ) {
                if (stack.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(stack.get(lastIndex));
                    stack.remove(lastIndex);
                }
            }

            else if ( input.equals("size") ) {
                System.out.println(stack.size());
            }

            else if ( input.equals("empty") ) {
                if (stack.isEmpty()) {
                    System.out.println(1);
                } else {
                    System.out.println(0);
                }
            }

            else if ( input.equals("top") ) {
                if (stack.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(stack.get(lastIndex));
                }
            }
        }
    }
}

File: 백준/Silver/10828. 스택/스택.kt
import java.util.Stack

fun main(args: Array<String>) {
    val n = readln().toInt()
    val stack = Stack<Int>()

    val commands = (0 until n)
        .map {
            readln()
        }

    val result: MutableList<Int> = mutableListOf()
    for (i in commands.indices) {
        val splitCommand = commands[i].split(" ")
        val commandPrefix = splitCommand[0]
        val commandPostfix = splitCommand.getOrNull(1)


        when (commandPrefix) {
            COMMAND.PUSH.prefix -> {
                val number = commandPostfix?.toInt() ?: throw RuntimeException()

                stack.push(number)
            }
            COMMAND.POP.prefix -> {
                result.add(stack.popOrNegativeNumber())
            }
            COMMAND.SIZE.prefix -> {
                result.add(stack.size)
            }
            COMMAND.EMPTY.prefix -> {
                result.add(stack.emptyReturnInt())
            }
            COMMAND.TOP.prefix -> {
                result.add(stack.topOrNegativeNumber())
            }
        }
    }

    val stringBuilder = java.lang.StringBuilder()

    result.forEach {
        stringBuilder.append("$it\n")
    }

    println(stringBuilder)
}

enum class COMMAND(
    val prefix: String
) {
    PUSH("push"),
    POP("pop"),
    SIZE("size"),
    EMPTY("empty"),
    TOP("top")
}

private fun Stack<Int>.popOrNegativeNumber(): Int {
    if (this.size == 0) return -1

    return this.pop()
}

private fun Stack<Int>.topOrNegativeNumber(): Int {
    if (this.size == 0) return -1

    return this.peek()
}

private fun Stack<Int>.emptyReturnInt(): Int {
    if (this.isEmpty()) return 1

    return 0
}